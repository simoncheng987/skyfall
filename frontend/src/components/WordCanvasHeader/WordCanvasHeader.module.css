.green {
  background-color: #b0da80;
}

.brown {
  background-color: #583b32;
}

.title {
  color: white;
  font-family: 'Poppins';
  font-weight: 900;
  font-size: 24px;
  -webkit-text-stroke: 1px black;
  text-shadow: 0px 2px 2px #1c1c1c;
}

.blueShadow {
  box-shadow: 0 10px 15px -8px rgba(42, 94, 172, 0.65);
}

.brownShadow {
  box-shadow: 0 10px 15px -8px rgba(101, 64, 53, 1);
}

.header {
  position: relative;
  border-radius: 15px 15px 0 0;
  padding: 10px 30px;
  display: flex;
  justify-content: space-between;
}
