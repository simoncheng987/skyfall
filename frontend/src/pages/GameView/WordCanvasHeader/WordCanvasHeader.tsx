import React from 'react';
import styles from './WordCanvasHeader.module.css';

interface WordCanvasHeaderProps {
  title: string;
  children?: React.ReactNode;
  className?: string;
  color?: 'green' | 'brown';
  shadowColor?: 'blue' | 'brown';
}

export default function WordCanvasHeader({
  title,
  children,
  className,
  color,
  shadowColor,
}: WordCanvasHeaderProps) {
  return (
    <div
      className={`${className} ${styles[color as string]} ${
        styles[`${shadowColor as string}Shadow`]
      } ${styles.header}`}
    >
      <span className={styles.title}>{title}</span>
      {children}
    </div>
  );
}

WordCanvasHeader.defaultProps = {
  children: null,
  className: '',
  color: 'green',
  shadowColor: 'blue',
};
