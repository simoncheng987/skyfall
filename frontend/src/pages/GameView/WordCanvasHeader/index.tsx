import WordCanvasHeader from './WordCanvasHeader';

export default WordCanvasHeader;
