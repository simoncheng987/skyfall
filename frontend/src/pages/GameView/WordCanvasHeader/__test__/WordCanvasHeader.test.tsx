import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import WordCanvasHeader from '../WordCanvasHeader';

describe('Word Canvas Header', () => {
  const title = 'testTitle';
  const children = 'testChildren';
  const className = 'testClassName';

  it('header with title only', () => {
    render(<WordCanvasHeader title={title} />);
    expect(screen.getByText(title)).toBeInTheDocument();
  });

  it('header with title and children', () => {
    render(<WordCanvasHeader title={title}>{children}</WordCanvasHeader>);

    expect(screen.getByText(title)).toBeInTheDocument();
    expect(screen.getByText(children)).toBeInTheDocument();
  });

  it('header with title and children', () => {
    render(<WordCanvasHeader title={title}>{children}</WordCanvasHeader>);

    expect(screen.getByText(title)).toBeInTheDocument();
    expect(screen.getByText(children)).toBeInTheDocument();
  });

  it('matches snapshot with no className prop', () => {
    const tree = renderer
      .create(<WordCanvasHeader title={title}>{children}</WordCanvasHeader>)
      .toTree();
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot with className prop', () => {
    const tree = renderer
      .create(
        <WordCanvasHeader title={title} className={className}>
          {children}
        </WordCanvasHeader>,
      )
      .toTree();
    expect(tree).toMatchSnapshot();
  });
});
