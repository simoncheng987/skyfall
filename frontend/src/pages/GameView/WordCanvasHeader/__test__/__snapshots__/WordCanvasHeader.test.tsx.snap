// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Word Canvas Header matches snapshot with className prop 1`] = `
Object {
  "instance": null,
  "nodeType": "component",
  "props": Object {
    "children": "testChildren",
    "className": "testClassName",
    "color": "green",
    "shadowColor": "blue",
    "title": "testTitle",
  },
  "rendered": Object {
    "instance": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="title"
        >
          testTitle
        </span>,
        "testChildren",
      ],
      "className": "testClassName green blueShadow header",
    },
    "rendered": Array [
      Object {
        "instance": null,
        "nodeType": "host",
        "props": Object {
          "children": "testTitle",
          "className": "title",
        },
        "rendered": Array [
          "testTitle",
        ],
        "type": "span",
      },
      "testChildren",
    ],
    "type": "div",
  },
  "type": [Function],
}
`;

exports[`Word Canvas Header matches snapshot with no className prop 1`] = `
Object {
  "instance": null,
  "nodeType": "component",
  "props": Object {
    "children": "testChildren",
    "className": "",
    "color": "green",
    "shadowColor": "blue",
    "title": "testTitle",
  },
  "rendered": Object {
    "instance": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="title"
        >
          testTitle
        </span>,
        "testChildren",
      ],
      "className": " green blueShadow header",
    },
    "rendered": Array [
      Object {
        "instance": null,
        "nodeType": "host",
        "props": Object {
          "children": "testTitle",
          "className": "title",
        },
        "rendered": Array [
          "testTitle",
        ],
        "type": "span",
      },
      "testChildren",
    ],
    "type": "div",
  },
  "type": [Function],
}
`;
